### ISSUE TITLE
[Bug]Authorize connection

### ISSUE THREAD

**Issue by sgdva** [jeswils-ap/anaplan-api#19]
Good day!
I'm running into an issue with the latest 0.2.10
I'm trying to do the auth but I get the following error

```
#trying new
import os
import anaplan
import anaplan.anaplan

TxtUserDownloadsFolder=os.path.join(os.path.expanduser("~"), "Downloads")
TxtUserDownloadsFolder=TxtUserDownloadsFolder.replace("\\", "/") +"/"
TxtCertPath=TxtUserDownloadsFolder + 'mycert.pem';TxtPrivateKeyPath=TxtUserDownloadsFolder + 'mykey.key'
auth = anaplan.anaplan.authorize("certificate",private_key=TxtPrivateKeyPath,certificate=TxtCertPath)
anaplan.anaplan.AnaplanConnection(auth,"82XXX","1243XXX")

TypeError                                 Traceback (most recent call last)
Cell In[2], line 9
      7 TxtUserDownloadsFolder=TxtUserDownloadsFolder.replace("\\", "/") +"/"
      8 TxtCertPath=TxtUserDownloadsFolder + 'mycert.pem';TxtPrivateKeyPath=TxtUserDownloadsFolder + 'mykey.key'
----> 9 auth = anaplan.anaplan.authorize("certificate",private_key=TxtPrivateKeyPath,certificate=TxtCertPath)
     10 anaplan.anaplan.AnaplanConnection(auth,"82XXX","1243XXX")

File c:\Users\User\AppData\Local\miniconda3\envs\myenv\Lib\site-packages\anaplan\anaplan.py:38, in authorize(method, **kwargs)
     29 def authorize(method: str, **kwargs) -> AuthorizationManager:
     30     """
     31     :param method: Authorization method
     32     :type method: str
   (...)
     36     :rtype: AuthorizationManager
     37     """
---> 38     return AuthorizationManager(method, **kwargs)

File c:\Users\User\AppData\Local\miniconda3\envs\myenv\Lib\site-packages\anaplan\authentication\AuthorizationManager.py:40, in AuthorizationManager.__init__(self, method, **kwargs)
     38 self.authorizer = AnaplanAuthentication()
     39 self.timer = None
---> 40 self.create_auth(method.lower(), **kwargs)
     41 self.start()

File c:\Users\User\AppData\Local\miniconda3\envs\myenv\Lib\site-packages\anaplan\authentication\AuthorizationManager.py:44, in AuthorizationManager.create_auth(self, method, **kwargs)
...
---> 22 required_params = set(auth_class.required_params)
     24 provided_params = set(kwargs.keys())
     25 missing_params = required_params - provided_params

TypeError: 'property' object is not iterable
```

Also it is worth to note that my privatekey is password protected, which I also encode for security, so I had to modify the calling function until sign_string to get this passcode.

In your previous version I updated for this with the following in the sign_string function, do you think you can append this as well?

```
def sign_string(message: bytes, priv_key: bytes,TxtPassword: str ="",TxtFilePathBin: str ="",TxtFilePathkey: str ="") -> str:
	#def sign_string(message: bytes, priv_key: bytes) -> str:
		"""Signs a string with a private key

		:param message: 150-character pseudo-random byte-array of characters
		:type message: bytes
		:param priv_key: Private key text, used to sign the message.
		:type priv_key: bytes
		:param TxtPassword: If the key is password protected and you have a string to encrypt it
		:type TxtPassword: string
		To create the Bin and key
		from cryptography.fernet import Fernet
		# Generate a key (do this only once and store it securely)
		key = Fernet.generate_key()
		# Save this key securely; you'll need it for decryption
		with open('C:/.../1.key', 'wb') as key_file:
			key_file.write(key)
		# Create a cipher instance
		cipher_suite = Fernet(key)
		# Your password to encrypt
		password = "your_password"
		# Encrypt the password
		encrypted_password = cipher_suite.encrypt(password.encode())
		# Write the encrypted password to a file
		with open('C:/.../2.bin', 'wb') as file_out:
			file_out.write(encrypted_password)
		# Load the key
		with open('C:/.../1.key', 'rb') as key_file:
			key = key_file.read()
		# Create a cipher instance
		cipher_suite = Fernet(key)
		# Read the encrypted password
		with open('C:/.../2.bin', 'rb') as file_in:
			encrypted_password = file_in.read()
		# Decrypt the password
		decrypted_password = cipher_suite.decrypt(encrypted_password).decode()
		# Use the decrypted password as needed
		:param TxtFilePathBin: If the key is password protected and you have the files with the encrypted one
		:type TxtPassword: string to .bin path
		:param TxtFilePathkey: If the key is password protected and you have the files with the encrypted one
		:type TxtPassword: string to .key path
		:raises ValueError: Error loading the private or signing the message.
		:return: Base64 encoded signed string value
		:rtype: str
		"""
		backend = default_backend()
		if TxtPassword!="": # 1. if TxtPassword!=""
			TxtDecodedPass=TxtPassword.encode()
		elif TxtFilePathBin!="" and TxtFilePathkey!="": # 1. if TxtPassword!=""
			try:
				with open(TxtFilePathkey, 'rb') as FileKey:
					Objkey = FileKey.read()
				cipher_suite = Fernet(Objkey)
				with open(TxtFilePathBin, 'rb') as file_in:
					Objbin = file_in.read()
				TxtDecodedPass = cipher_suite.decrypt(Objbin).decode().encode()
			except Exception as TxtException:
				logger.error(f"Error loading private key {TxtException}", exc_info=True)
				raise ValueError(f"Error loading private key {TxtException}")
		else: # 1. if TxtPassword!=""
			TxtDecodedPass=None
		try:
			if not os.path.isfile(priv_key):
				key = serialization.load_pem_private_key(priv_key, TxtDecodedPass, backend=backend)
			else:
				with open(priv_key, 'r') as key_file:
					key = serialization.load_pem_private_key(open(priv_key, 'r').read().encode('utf-8'), TxtDecodedPass, backend=backend)
			try:
				signature = key.sign(message, padding.PKCS1v15(), hashes.SHA512())
				return b64encode(signature).decode('utf-8')
			except ValueError as e:
				logger.error(f"Error signing message {e}", exc_info=True)
				raise ValueError(f"Error signing message {e}")
		except ValueError as e:
			logger.error(f"Error loading private key {e}", exc_info=True)
			raise ValueError(f"Error loading private key {e}")
```

**Comment by jeswils-ap** [comment]
This bug is triggered by handling of Python class properties. Changed to directly referencing the required_params variable instead of using the property decorator.

I'll open a new issue for the request to enable support for password encoded keys as a feature enhancement. For now, Java Keystore support is working as expected. Instructions for creating this with Anaplan Certificate authentication are found in the Anaplan Connect guide.
